### PR TITLE
Make delimiters separate tokens from beginnings of '%' literals

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
This reduces the size of the parser object file from about 14mb to about 9mb. It does allow comments and whitespace between `%q` and its delimiter, but I think it is ok in general to parse a superset of the language.

/cc @robrix

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -339,7 +356,11 @@ module.exports = grammar({
 function stringBody (open, close, insert) {
   var contents = [ /\\./, RegExp('[^\\\\\\' + close + ']') ];
   if (typeof insert !== 'undefined') contents.push(insert);
-  return seq(open, repeat(choice.apply(null, contents)), token(prec(PREC.LITERAL, close)));
+  return seq(
+    (typeof open === 'string') ? token(prec(PREC.LITERAL, open)) : open,
```
Comment (maxbrunsfeld) [inline grammar.js]: This ternary is a temporary hack to work around a tree-sitter bug when handling `token(blank())`.

File: grammar.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const PREC = {
   LITERAL: 100,
 };
 
-const unbalancedDelimiters = '!@#$%^&*)]}>|\\=/+-~`\'",.?:;_';
+const unbalancedDelimiters = '!@#$%^&*)]}>|\\=/+-~`\'",.?:;_'.split('');
```
Comment (robrix) [inline grammar.js]: Heh, good call.

File: grammar_test/literals.txt [hunk 3]
```diff
@@ -288,37 +288,10 @@ percent q string with unbalanced delimiters
...
-%q'a'
-%q"a"
-%q)a)
-%q]a]
-%q}a}
-%q>a>
-%q`a`
-%q~a~
-%q_a_
-%q#a#
```
Comment (robrix) [inline grammar_test/literals.txt]: Any chance you could restore these? I’m concerned about the `%q#a#` one in particular.

**Comment by robrix** [comment]
:bow:

File: grammar_test/literals.txt (hunk 3 above)
Comment (maxbrunsfeld) [inline grammar_test/literals.txt]: Oh haha sorry. Didn't mean to leave that like that.

**Comment by robrix** [comment]
Nice one! Thank you!

**Pull request merged**